src/api/__tests__/twitter.test.js

```diff
@@ -0,0 +1,103 @@
+const TwitterHandler = require('../twitter')
+
+describe('TwitterHandler', () => {
+    let sut
+    let twitterMgrMock = {
+        findDidInTweets: jest.fn()
+    }
+    let claimMgrMock  ={
+        issue: jest.fn()
+    }
+
+
+    beforeAll(() => {
+        sut = new TwitterHandler(twitterMgrMock, claimMgrMock)
+    })
+
+    test('empty constructor', () => {
+        expect(sut).not.toBeUndefined()
+    })
+
+    test('handle null body', done => {
+        sut.handle({}, {}, (err, res) => {
+            expect(err).not.toBeNull()
+            expect(err.code).toEqual(400)
+            expect(err.message).toBeDefined()
+            done()
+        })
+    })
+
+    test('not coming from the 3box origin', done => {
+        sut.handle({ headers: {Origin: "abc"}, body: '{}' }, {}, (err, res) => {
+            expect(err).not.toBeNull()
+            expect(err.message).toEqual('unauthorized')
+            expect(err.code).toEqual(401)
+            done()
+        })
+    })
+
+    test('no did', done => {
+        sut.handle(
+            { headers: {Origin: "Origin: https://3box.io"}, body: JSON.stringify({ other: 'other' }) },
```

Comment:
Not sure that this is correct. Chrome and FF seems to be sending `{ origin: https://3box.io }` as a header for example.